src/NumericalAlgorithms/DiscontinuousGalerkin/NumericalFluxes/HartenLaxVanLeer.hpp

```diff
@@ -0,0 +1,211 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace dg {
+namespace NumericalFluxes {
+
+/*!
+ * \ingroup DiscontinuousGalerkinGroup
+ * \ingroup NumericalFluxesGroup
+ * \brief Compute the Harten-Lax-van Leer numerical flux.
+ *
+ * Let \f$U\f$ and \f$F^i(U)\f$ be the state vector of the system and its
+ * corresponding volume flux, respectively. Let \f$n_i\f$ be the unit normal to
+ * the interface. Defining \f$F_n(U) := n_i F^i(U)\f$, and denoting the
+ * corresponding projections of the numerical fluxes as \f${F_n}^*(U)\f$, the
+ * Harten-Lax-van Leer flux for each variable is
+ *
+ * \f{align*}
+ * {F_n}^*(U) = \frac{c_\text{max} F_n(U_\text{int}) -
+ * c_\text{min} F_n(U_\text{ext})}{c_\text{max} - c_\text{min}}
+ * - \frac{c_\text{min}c_\text{max}}{c_\text{max} - c_\text{min}}
+ * \left(U_\text{int} - U_\text{ext}\right)
+ * \f}
+ *
+ * where "int" and "ext" stand for interior and exterior, respectively.
+ * \f$c_\text{min}\f$ and \f$c_\text{max}\f$ are estimates on the minimum
```

Comment:
Where do these estimates come from? How good are the estimates? Is there a reference that discusses this?

Reply:
The HLL flux requires estimated wave speeds (...if you knew the solution to the Riemann problem, you wouldn't be using an approximate Riemann solver) as inputs. I suggest adding a citation (in doxygen) to the HLL reference
https://doi.org/10.1137/1025002
and the more modern textbook
https://doi.org/10.1007/b79761